plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/CrowdFunding.hs

```diff
@@ -244,4 +239,5 @@ successfulCampaign =
         >> makeContribution (Trace.Wallet 4) (Ada.lovelaceValueOf 1)
         >> Trace.addBlocks 18
         >> Trace.notifySlot (Trace.Wallet 1)
+        >> Trace.handleUtxoQueries (Trace.Wallet 1)
```

Comment:
Isn't this done by the next line?

Reply:
Yeah but the order is wrong (the next line processes unbalanced transactions first, and then responds to UTXO queries). I'll just take `handleUtxoQueries` out of `handleBlockchainEvents`.

Reply:
Then we need to call it every time we call `handleBlockchainEvnets`, which seems annoying... It seems nicer for the traces to have one "process everything" event.

Reply:
Okay, that makes sense, I think I need to reorganise `handleBlockchainEvents` a little.

Reply:
I put it back. But I don't think there's a way around having to call `handleBlockchainEvents` twice, because the second blockchain event (writing an `UnbalancedTx` that collects the contributions) happens after the first one, not in parallel.